tests/System.IO.Pipelines.Tests/PipeCompletionCallbacksTests.cs

```diff
@@ -0,0 +1,453 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class PipeCompletionCallbacksTests
+    {
+        private readonly PipeFactory _pipeFactory;
+
+        public PipeCompletionCallbacksTests()
+        {
+            _pipeFactory = new PipeFactory();
+        }
+
+        public void Dispose()
+        {
+            _pipeFactory.Dispose();
+        }
+
```

Comment:
This is not recommended use case because it can hit "Cannot complete during writing" but I will add a test